dev/set_matrix.py

```diff
@@ -368,19 +388,92 @@ def get_flavor(name):
     return {"pytorch-lightning": "pytorch"}.get(name, name)
 
 
+def validate_test_coverage(flavor: str, config: Dict):
+    """
+    Validate that all test files for the flavor are executed in the cross-version tests.
+
+    This is done by parsing `run` commands in the `ml-package-versions.yml` to get the list
+    of executed test files, and then comparing it with the actual test files in the directory.
+    """
+    test_dir = os.path.join("tests", flavor)
+    tested_files = set()
+
+    for category, cfg in config.categories:
+        if not cfg.run:
+            continue
+
+        # Consolidate multi-line commands with "\" to a single line
+        commands = []
+        curr = ""
+        for cmd in cfg.run.split("\n"):
+            if cmd.endswith("\\"):
+                curr += cmd.rstrip("\\")
+            else:
+                commands.append(curr + cmd)
+                curr = ""
+
+        # Parse pytest commands to get the executed test files
+        for cmd in commands:
+            cmd = cmd.strip().rstrip(";")
+            if cmd.startswith("pytest"):
+                tested_files |= _get_test_files_from_pytest_command(cmd, test_dir)
+
+    if untested_files := _get_test_files(test_dir) - tested_files:
+        # TODO: Update this after fixing ml-package-versions.yml to
+        # have all test files in the matrix.
+        warnings.warn(
+            f"Flavor '{flavor}' has test files that are not covered by the test matrix. \n"
+            + "\n".join(f" - {t}" for t in untested_files)
+            + f"\nPlease update {VERSIONS_YAML_PATH} to execute all test files."
+        )
```

Comment:
Q:
This might miss some cases, if the script generates test file name in script, e.g.

```
for filepath in path/to/*.py; do  # e.g. we can append some filter condition here
  pytest filename
```

is it possible to detect this case ?

Reply:
> is it possible to detect this case ?

No, but do we need to handle it? I think it is generally better to keep the `run` command simple.

Reply:
agreed it is too complex to support it and it is unnecessary currently.
can we add this limitation into the L429 warning message ? so that other developer can be aware of this limitation.

Reply:
> can we add this limitation into the L429 warning message ?

100% makes sense! Will add:)